src/main/resources/data/computercraft/lua/rom/modules/main/cc/expect.lua

```diff
@@ -88,7 +88,30 @@ local function field(tbl, index, ...)
     end
 end
 
+--- Expect a number to be within a specific range.
+--
+-- @tparam number num, The value to check.
+-- @tparam min The minimum value, if nil then `-math.huge` is used.
+-- @tparam max The maximum value, if nil then `math.huge` is used.
+-- @return The given `value`.
+-- @throws If the value is outside of the allowed range.
+local function range(num, min, max)
+  expect(1, num, "number")
+  local min = expect(2, min, "number", "nil") or -math.huge
+  local max = expect(3, max, "number", "nil") or math.huge
+  if min > max then
+      error("min must be less than or equal to max)", 2)
+  end
+  
+  if num < min or num > max then
```

Comment:
```suggestion
  if num ~= num or num < min or num > max then
```

This way we also catch when `num` is NaN. Might be worth having a test for this too :/.

Reply:
I feel like being a bit more explicit that we are testing for NaN would be better, which might be best done but adding a small function for that.

```lua
local function isNan(num)
  return num ~= num
end
```